SEED-platform/seed
# Fix bug where compliance calculations were not being cast to the right data types

#### Any background context you want to provide?
Compliance calculations were not casting the data types before checking causing the front end to not show data.

#### What's this PR do?
* Move the `DATA_TYPE_PARSERS` that was in `search.py` to the Column class. 
* Adds a class method and instance method to cast data from the column. It is only one value at a time at the moment
* Cast `date` as `date` instead of `datetime` (which was happening)
* Write a bunch of test cases for casting data
* Update the pre-commit versions

#### How should this be manually tested?
* Import program target data as a string. This may not be possible from the UI, but it did happen using py-seed
* Set the program metric up
* View the compliance page
* Also, monkey test the search, but the unit tests should cover that

#### What are the relevant tickets?
#3788 

#### Screenshots (if appropriate)
We can now view lots of data in the program insights page!
![image](https://user-images.githubusercontent.com/1907354/214474545-3bcef700-4572-45ca-bf0e-a00e55c03774.png)
